lxd/cluster/membership.go

```diff
@@ -958,8 +957,8 @@ func Purge(cluster *db.Cluster, name string) error {
 			return errors.Wrapf(err, "Failed to remove member %q", name)
 		}
 
-		filter := db.CertificateFilter{Name: name, Type: db.CertificateTypeServer}
-		err = tx.DeleteCertificates(filter)
+		certType := db.CertificateTypeServer
+		err = tx.DeleteCertificatesByNameAndType(name, certType, db.CertificateFilter{})
```

Comment:
@masnax this is great, this is the experience that I was hoping for and it will be a lot harder to mess up accidentally. :)